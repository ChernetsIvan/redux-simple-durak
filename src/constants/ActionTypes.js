export const START_GAME_CLICKED = 'START_GAME_CLICKED';
export const BEGIN_GAME_CLICKED = 'BEGIN_GAME_CLICKED';
