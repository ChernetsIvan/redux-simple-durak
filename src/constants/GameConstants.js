export const cardRanks = [
  { text: "6", cardValue: 1 },
  { text: "7", cardValue: 2 },
  { text: "8", cardValue: 3 },
  { text: "9", cardValue: 4 },
  { text: "10", cardValue: 5 },
  { text: "В", cardValue: 6 },
  { text: "Д", cardValue: 7 },
  { text: "К", cardValue: 8 },
  { text: "Т", cardValue: 9 }
];

export const cardSuits = [
  { suit: "Ч" },
  { suit: "К" },
  { suit: "П" },
  { suit: "Б" }
];

export const statusMessages = {
  PlayerAttack: "Ваш ход!",
  PlayerDefence: "Защищайтесь!",
  PlayerDiscard: "Беру! Подбрасывать будете?",
  PlayerLose: "Вы проиграли! Сыграем ещё раз ? :)",
  PlayerWin: "Вы выиграли! Поздравляю :)",
  GameResultDraw: "Ничья!"
};
