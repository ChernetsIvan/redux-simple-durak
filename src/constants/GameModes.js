export const PlayerAttack = "PlayerAttack";
export const PlayerDefence = "PlayerDefence";
export const PlayerDiscard = "PlayerDiscard"; //AI забирает: игрок подбрасывает ему карты
export const AiAttack = "AiAttack";
export const AiDefence = "AiDefence";
export const AiDiscard = "AiDiscard"; //Игрок забирает: AI подбрасывает ему карты

export const PlayerWin = "PlayerWin";
export const AIWin = "AIWin";
export const TheDraw = "TheDraw"; //Ничья