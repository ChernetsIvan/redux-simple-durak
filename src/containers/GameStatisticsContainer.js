import { connect } from "react-redux";

import { GameStatistics } from "./../components/GameStatistics";

import { clickOnBeginGameButton } from "./../actions";

const mapStateToProps = store => ({
  playerWins: store.statistics.playerWins,
  aiWins: store.statistics.aiWins,
  draws: store.statistics.draws
});

const mapDispatchToProps = dispatch => ({
  onBeginGameClick: () => {
    dispatch(clickOnBeginGameButton());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(GameStatistics);
