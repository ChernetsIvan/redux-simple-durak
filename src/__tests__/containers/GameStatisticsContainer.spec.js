import React from "react";
import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { createStore } from "redux";

import GameStatisticsContainer, {
  mapDispatchToProps
} from "./../../containers/GameStatisticsContainer";
import reducer from "./../../reducers";
import * as actionTypes from "./../../constants/ActionTypes";
import { AIWin, PlayerWin, TheDraw } from "./../../constants/GameModes";

describe("GameStatisticsContainer snapshot testing", () => {
  let store;

  beforeEach(() => {
    store = createStore(reducer);
  });

  test("Dispatch with GAME_OVER-action, AI win", () => {
    store.dispatch({
      type: actionTypes.GAME_OVER,
      payload: AIWin
    });
    const component = renderer.create(
      <Provider store={store}>
        <GameStatisticsContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Store in initial-state mode", () => {
    const component = renderer.create(
      <Provider store={store}>
        <GameStatisticsContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Dispatch with GAME_OVER-action, Player win", () => {
    store.dispatch({
      type: actionTypes.GAME_OVER,
      payload: PlayerWin
    });
    const component = renderer.create(
      <Provider store={store}>
        <GameStatisticsContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Dispatch with GAME_OVER-action, the draw", () => {
    store.dispatch({
      type: actionTypes.GAME_OVER,
      payload: TheDraw
    });
    const component = renderer.create(
      <Provider store={store}>
        <GameStatisticsContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Dispatch with GAME_OVER-action, incorrect payload", () => {
    store.dispatch({
      type: actionTypes.GAME_OVER,
      payload: "IncorrectPayload"
    });
    const component = renderer.create(
      <Provider store={store}>
        <GameStatisticsContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  /*
  test("Checking mapDispatchToProps-function", () => {
    store.dispatch({
      type: actionTypes.BEGIN_GAME_CLICKED
    });
    const action = store.getActions();
    expect(action[0].type).toBe("BEGIN_GAME_CLICKED");
  });
  */
});

describe("Map Dispatch To Props", () => {
  test("should call onBeginGameClick action", () => {
    const mockFunction = jest.fn(() => 'Mock function');
    const { onBeginGameClick } = mapDispatchToProps(mockFunction);
    onBeginGameClick();
    expect(mockFunction).toBeCalled();
  });
});
