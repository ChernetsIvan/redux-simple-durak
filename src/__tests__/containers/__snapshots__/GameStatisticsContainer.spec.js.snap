// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameStatisticsContainer snapshot testing Dispatch with GAME_OVER-action, AI win 1`] = `
Object {
  "className": "container",
}
`;

exports[`GameStatisticsContainer snapshot testing Dispatch with GAME_OVER-action, Player win 1`] = `
Object {
  "className": "container",
}
`;

exports[`GameStatisticsContainer snapshot testing Dispatch with GAME_OVER-action, incorrect payload 1`] = `
Object {
  "className": "container",
}
`;

exports[`GameStatisticsContainer snapshot testing Dispatch with GAME_OVER-action, the draw 1`] = `
Object {
  "className": "container",
}
`;

exports[`GameStatisticsContainer snapshot testing Store in initial-state mode 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center mb-3 mt-5"
  >
    <div
      className="col-auto"
    >
      <h2>
        Статистика
      </h2>
    </div>
  </div>
  <div
    className="row justify-content-center mb-5"
  >
    <div
      className="col-6"
    >
      <h4
        className="text-right"
      >
        Побед:
      </h4>
      <h4
        className="text-right"
      >
        Ничьих:
      </h4>
      <h4
        className="text-right"
      >
        Поражений:
      </h4>
      <h4
        className="text-right"
      >
        Всего сыграно, игр:
      </h4>
    </div>
    <div
      className="col-6"
    >
      <h4>
        0
      </h4>
      <h4>
        0
      </h4>
      <h4>
        0
      </h4>
      <h4>
        0
      </h4>
    </div>
  </div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-auto"
    >
      <a
        onClick={[Function]}
        style={Object {}}
      >
        Новая игра
      </a>
    </div>
  </div>
</div>
`;
